src/Microsoft.DotNet.Core.Build.Tasks/GenerateRuntimeConfigurationFiles.cs

```diff
@@ -0,0 +1,128 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using Microsoft.DotNet.ProjectModel;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using NuGet.ProjectModel;
+
+namespace Microsoft.DotNet.Cli.Tasks
+{
+    /// <summary>
+    /// Generates the $(project).runtimeconfig.json and optionally $(project).runtimeconfig.dev.json files
+    /// for a project.
+    /// </summary>
+    public class GenerateRuntimeConfigurationFiles : Task
+    {
+        [Required]
+        public string RuntimeOutputPath { get; set; }
+
+        [Required]
+        public string OutputName { get; set; }
+
+        [Required]
+        public string LockFilePath { get; set; }
+
+        public string RawRuntimeOptions { get; set; }
+
+        public bool IncludeDevConfig { get; set; }
+
+        private LockFile LockFile { get; set; }
+
+        public override bool Execute()
+        {
+            LockFile = LockFileCache.Instance.GetLockFile(LockFilePath);
+
+            WriteRuntimeConfig();
+
+            if (IncludeDevConfig)
+            {
+                WriteDevRuntimeConfig();
+            }
+
+            return true;
+        }
+
+        private void WriteRuntimeConfig()
+        {
+            var json = new JObject();
```

Comment:
I agree with Bryan, you should be able to simply set your ProjectLock object properties and then do a JsonConvert.SerializeObject(projectLock);

Also, this would make testing this a lot easier as we could simply create a project lock POCO to pass around.